### PR TITLE
Refactor tests when moving to k8s.io/client-go v0.26.0

### LINKED ISSUE #4013: Replace `k8s.io/client-go/scale` with controller-runtime client
### Proposal

controller-runtime client will soon support accessing `/scale` subresource:
https://github.com/kubernetes-sigs/controller-runtime/pull/2094
https://github.com/kubernetes-sigs/controller-runtime/issues/172

Once a version of controller-runtime with this change is released we should use it and consolidate our codebase ->
replace `k8s.io/client-go/scale` with controller-runtime client.

It is in the ScaledObject controller and scale executor:
https://github.com/kedacore/keda/blob/f4bee67f5a444338b7c2b93c7b762a27c348867b/controllers/keda/scaledobject_controller.go#L67
https://github.com/kedacore/keda/blob/f4bee67f5a444338b7c2b93c7b762a27c348867b/pkg/scaling/executor/scale_executor.go#L47

### PR DESCRIPTION
**Pull request by dttung2905**
Hi team,
This is the first PR to resolve #4013. There will be a followed up PR on this issue as this is a relatively big one and I would like to break it down into smaller pieces

A few things I have done:
- Migrate some failed test when moving to `k8s.io/client-go v0.26.0` as it uses ginkgov2 ([release note](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.14.0))
- Migrate async testing for set up node with `BeforeSuite()`. I take references from here https://github.com/onsi/ginkgo/issues/882. Although I think we might need to refactor even further but I would leave that for another MR
- When upgrading, I also faced an exact bug like https://github.com/kubernetes-sigs/controller-tools/issues/757. Temporarily I add a Kustomize patch under JsonPatches6902. We can revisit this once k8s 1.26.1 is released

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -48,10 +49,9 @@ require (
 	github.com/microsoft/azure-devops-go-api/azuredevops v1.0.0-b5
 	github.com/mitchellh/hashstructure v1.1.0
 	github.com/newrelic/newrelic-client-go v1.1.0
-	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/ginkgo/v2 v2.5.0
-	github.com/onsi/gomega v1.24.1
-	github.com/open-policy-agent/cert-controller v0.6.0
+	github.com/onsi/ginkgo/v2 v2.6.1
+	github.com/onsi/gomega v1.24.2
+	github.com/open-policy-agent/cert-controller v0.5.0
```
Comment (JorTurFer) [inline go.mod:54]: Don't downgrade this package, we need latest version of it for the certificate generation

File: controllers/keda/suite_test.go [hunk 2]
```diff
@@ -22,21 +22,19 @@ import (
 	"testing"
 	"time"
 
-	. "github.com/onsi/ginkgo"
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/k8s"
+	"github.com/kedacore/keda/v2/pkg/scaling"
```
Comment (JorTurFer) [inline controllers/keda/suite_test.go:27]: There is a style rule for imports (and it's detected by static checks).
Imports have to be split in 3 groups, the first has to be go imports, the second one has to be the external packages and the last one has to be internal packages. 
Move this 3 imports to the end and split them in a group. Pleas, also join 
```
logf "sigs.k8s.io/controller-runtime/pkg/log"
	"sigs.k8s.io/controller-runtime/pkg/log/zap"
```
with the second group

File: go.mod (hunk 1 above)
Comment (dttung2905) [reply go.mod]: Yes. I just saw the CI failure this morning. I have already made the changes

File: pkg/metricsservice/api/metrics.pb.go [hunk 3]
```diff
@@ -16,7 +16,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.28.1
-// 	protoc        v3.21.12
+// 	protoc        v3.6.1
```
Comment (zroubalik) [inline pkg/metricsservice/api/metrics.pb.go:19]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

File: pkg/metricsservice/api/metrics_grpc.pb.go [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go-grpc. DO NOT EDIT.
 // versions:
 // - protoc-gen-go-grpc v1.2.0
-// - protoc             v3.21.12
+// - protoc             v3.6.1
```
Comment (zroubalik) [inline pkg/metricsservice/api/metrics_grpc.pb.go:4]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

File: pkg/scalers/externalscaler/externalscaler.pb.go [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.28.1
-// 	protoc        v3.21.12
+// 	protoc        v3.6.1
```
Comment (zroubalik) [inline pkg/scalers/externalscaler/externalscaler.pb.go:4]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

File: pkg/scalers/externalscaler/externalscaler_grpc.pb.go [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go-grpc. DO NOT EDIT.
 // versions:
 // - protoc-gen-go-grpc v1.2.0
-// - protoc             v3.21.12
+// - protoc             v3.6.1
```
Comment (zroubalik) [inline pkg/scalers/externalscaler/externalscaler_grpc.pb.go:4]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

File: pkg/scalers/liiklus/LiiklusService.pb.go [hunk 7]
```diff
@@ -1,16 +1,16 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
```
Comment (zroubalik) [inline pkg/scalers/liiklus/LiiklusService.pb.go:2]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

File: pkg/scalers/liiklus/LiiklusService_grpc.pb.go [hunk 8]
```diff
@@ -1,17 +1,17 @@
 // Code generated by protoc-gen-go-grpc. DO NOT EDIT.
 // versions:
 // - protoc-gen-go-grpc v1.2.0
```
Comment (zroubalik) [inline pkg/scalers/liiklus/LiiklusService_grpc.pb.go:3]: please remove this from the PR, it is not relevant and we are downgrading the protoc version

**Review by zroubalik — COMMENTED** [review]
Thank you for doing this! 

Please also see the failing Static Checks, there are a few suggestions how to modify the code.

**Pull request merged**